staging/src/k8s.io/client-go/dynamic/fake/simple.go

```diff
@@ -116,10 +121,23 @@ func NewSimpleDynamicClientWithCustomListKinds(scheme *runtime.Scheme, gvrToList
 	cs.AddWatchReactor("*", func(action testing.Action) (handled bool, ret watch.Interface, err error) {
 		gvr := action.GetResource()
 		ns := action.GetNamespace()
+
+		if replable, ok := o.(ReplayableWatch); ok {
+			if watchAction, ok := action.(testing.WatchAction); ok && watchAction.GetWatchRestrictions().RequiresInitialEvents {
+				klog.Infof("replaying initial events")
+				watch, err := replable.WatchWithReplay(gvr, ns)
+				if err != nil {
+					return false, nil, err
+				}
+				return true, watch, nil
+			}
+		}
+
```

Comment:
This implementation is for the dynamic client, but this bug occurs also in the generated native clientsets. Waiting on answer about WatchWithReplay vs Watch new argument to refactor